presto-docs/src/main/sphinx/connector/oracle.rst

```diff
@@ -23,47 +20,362 @@ connection properties as appropriate for your setup:
+To see a list of the columns in the ``clicks`` table in the ``web``
+database run either of the following::
 
     DESCRIBE oracle.web.clicks;
     SHOW COLUMNS FROM oracle.web.clicks;
 
-Finally, you can access the ``clicks`` table in the ``web`` database::
+To access the ``clicks`` table in the ``web`` database run::
 
     SELECT * FROM oracle.web.clicks;
 
-If you used a different name for your catalog properties file, use
-that catalog name instead of ``oracle`` in the above examples.
+Mapping data types between Presto and Oracle
+--------------------------------------------
+
+Oracle supports types that Presto doesn't and Presto supports types that Oracle
+doesn't. The following sections explain their type mapping.
+
+Oracle to Presto type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports selecting Oracle database types. This table shows the Oracle to
+Presto data type mapping:
+
+
+.. list-table:: Oracle to Presto type mapping
+  :widths: 20, 20, 60
+  :header-rows: 1
+
+  * - Oracle database type
+    - Presto type
+    - Notes
+  * - ``NUMBER(p, s)``
+    - ``DECIMAL(p, s)``
+    -  See :ref:`number mapping`
+  * - ``NUMBER(p)``
+    - ``DECIMAL(p, 0)``
+    - See :ref:`number mapping`
+  * - ``FLOAT[(p)]``
+    - ``DOUBLE``
+    -
+  * - ``BINARY_FLOAT``
+    - ``REAL``
+    -
+  * - ``BINARY_DOUBLE``
+    - ``DOUBLE``
+    -
+  * - ``VARCHAR2(n CHAR)``
+    - ``VARCHAR(n)``
+    -
+  * - ``VARCHAR2(n BYTE)``
+    - ``VARCHAR(n)``
+    -
+  * - ``NVARCHAR2(n)``
+    - ``VARCHAR(n)``
+    -
+  * - ``CHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``NCHAR(n)``
+    - ``CHAR(n)``
+    -
+  * - ``CLOB``
+    - ``VARCHAR``
+    -
+  * - ``NCLOB``
+    - ``VARCHAR``
+    -
+  * - ``RAW(n)``
+    - ``VARBINARY``
+    -
+  * - ``BLOB``
+    - ``VARBINARY``
+    -
+  * - ``DATE``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p)``
+    - ``TIMESTAMP``
+    - See :ref:`datetime mapping`
+  * - ``TIMESTAMP(p) WITH TIME ZONE``
+    - ``TIMESTAMP WITH TIME ZONE``
+    - See :ref:`datetime mapping`
+
+If an Oracle table contains a type that's not listed in the above table, an
+error shows:
+
+ - An ``unsupported-type.handling-strategy`` equals ``FAIL``
+   (default configuration), any queries on the table fail.
+ - An ``unsupported-type.handling-strategy`` equals ``IGNORE``
+   (default configuration), Presto no longer accesses the column.
+ - An ``unsupported-type.handling-strategy`` equals ``CONVERT_TO_VARCHAR``
+   (default configuration), a column value converts to an unbounded ``VARCHAR``.
+
+Presto to Oracle type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Presto supports creating tables with the following types in an Oracle database.
+The table shows the mappings from Presto to Oracle data types:
+
+.. note::
+   For types not listed in the table below, Presto can't perform the ``CREATE
+   TABLE`` and ``AS SELECT``operations to insert info into existing
+   tables that use ``Oracle to Presto`` type mapping.
```

Comment:
this is single operation `CREATE TABLE <some_table> AS SELECT ....`

```suggestion
   For types not listed in the table below, Presto can't perform the ``CREATE
   TABLE <table> AS SELECT`` operations. 
   When data is inserted into existing tables ``Oracle to Presto`` type mapping is used.
```

Reply:
Added to edits.